scheds/rust/scx_lavd/src/bpf/sys_stat.bpf.c

```diff
@@ -63,10 +64,66 @@ static void init_sys_stat_ctx(struct sys_stat_ctx *c)
 	c->stat_next->last_update_clk = c->now;
 }
 
-static void collect_sys_stat(struct sys_stat_ctx *c)
+static void plan_x_cpdom_migration(struct sys_stat_ctx *c)
 {
+	struct cpdom_ctx *cpdomc;
 	u64 dsq_id;
-	int cpu, nr;
+	u32 avg_nr_q_tasks_per_cpu = 0, nr_q_tasks, x_mig_delta;
+	u32 stealer_threshold, stealee_threshold;
+
+	/*
+	 * Calcualte average queued tasks per CPU per compute domain.
+	 */
+	bpf_for(dsq_id, 0, nr_cpdoms) {
+		if (dsq_id >= LAVD_CPDOM_MAX_NR)
+			break;
+
+		nr_q_tasks = scx_bpf_dsq_nr_queued(dsq_id);
+		c->nr_queued_task += nr_q_tasks;
+
+		cpdomc = MEMBER_VPTR(cpdom_ctxs, [dsq_id]);
+		cpdomc->nr_q_tasks_per_cpu = (nr_q_tasks * 1000) / cpdomc->nr_cpus;
```

Comment:
Small question, why do we need `* 1000` here? guessing you want to keep the value to 2 decimal places?
`avg_nr_q_tasks_per_cpu >> LAVD_CPDOM_MIGRATION_SHIFT` is going be performed, wouldn't `nr_q_tasks << LAVD_CPDOM_MIGRATION_SHIFT` be more intuitive ?

Reply:
Yes, it is to keep 2 decimal places for better accuracy. LAVD_CPDOM_MIGRATION_SHIFT is to maintain +/- 12.5% delta.